### ISSUE TITLE
Unable to test successfully

### ISSUE THREAD

**Issue by simon28li** [zeromq/pyzmq#1740]
hi，When I tested version 23.2.0, I encountered the following error.
```
zmq/tests/test_mypy.py::test_mypy_example[asyncio] FAILED                                                                                                                             [ 64%]
zmq/tests/test_mypy.py::test_mypy_example[chat] FAILED                                                                                                                                [ 64%]
zmq/tests/test_mypy.py::test_mypy_example[cython] FAILED                                                                                                                              [ 64%]
zmq/tests/test_mypy.py::test_mypy_example[device] FAILED                                                                                                                              [ 64%]
zmq/tests/test_mypy.py::test_mypy_example[draft] FAILED                                                                                                                               [ 65%]
zmq/tests/test_mypy.py::test_mypy_example[eventloop] FAILED                                                                                                                           [ 65%]
zmq/tests/test_mypy.py::test_mypy_example[gevent] FAILED                                                                                                                              [ 65%]
zmq/tests/test_mypy.py::test_mypy_example[heartbeat] FAILED                                                                                                                           [ 65%]
zmq/tests/test_mypy.py::test_mypy_example[logger] FAILED                                                                                                                              [ 65%]
zmq/tests/test_mypy.py::test_mypy_example[mongodb] FAILED                                                                                                                             [ 66%]
zmq/tests/test_mypy.py::test_mypy_example[monitoring] FAILED                                                                                                                          [ 66%]
zmq/tests/test_mypy.py::test_mypy_example[poll] FAILED                                                                                                                                [ 66%]
zmq/tests/test_mypy.py::test_mypy_example[pubsub] FAILED                                                                                                                              [ 66%]
zmq/tests/test_mypy.py::test_mypy_example[security] FAILED                                                                                                                            [ 67%]
zmq/tests/test_mypy.py::test_mypy_example[serialization] FAILED                                                                                                                       [ 67%]
zmq/tests/test_mypy.py::test_mypy_example[win32-interrupt] FAILED                                                                                                                     [ 67%]
zmq/tests/test_mypy.py::test_mypy[test_context.py] FAILED                                                                                                                             [ 67%]
zmq/tests/test_mypy.py::test_mypy[test_socket.py] FAILED                                                                                                                              [ 68%]
zmq/tests/test_mypy.py::test_mypy[test_toplevel.py] FAILED
```
Specific details of reporting errors.
```
 def run_mypy(*mypy_args):
        """Run mypy for a path

        Captures output and reports it on errors
        """
        p = Popen(
            [sys.executable, "-m", "mypy"] + list(mypy_args), stdout=PIPE, stderr=STDOUT
        )
        o, _ = p.communicate()
        out = o.decode("utf8", "replace")
        print(out)
>       assert p.returncode == 0, out
E       AssertionError: zmq/sugar/socket.py:367: error: Call to untyped function "SocketOption" in typed context
E         zmq/sugar/socket.py:853: error: Call to untyped function "_deserialize" in typed context
E         zmq/sugar/socket.py:894: error: Call to untyped function "_deserialize" in typed context
E         zmq/sugar/socket.py:936: error: Call to untyped function "_deserialize" in typed context
E         Found 4 errors in 1 file (checked 1 source file)
E
E       assert 1 == 0
E        +  where 1 = <subprocess.Popen object at 0xffff8d09f5b0>.returncode

zmq/tests/test_mypy.py:47: AssertionError
```
Can you give me some help?

**Comment by minrk** [comment]
Can you share more environment info (e.g. `pip freeze` and platform / version info)?

**Comment by minrk** [comment]
This was apparently a change in recent mypy, and fixed by #1747
